### PR TITLE
Make GitHub Actions Auto-style Correctly

### PR DESCRIPTION
**Pull request by kfu02**
## Description

This PR improves upon #2015 by making automatic style formatting happen in a separate PR, which tags the original PR author to review and merge it. That way, on merge, the other automated checks will run after style fixes are automatically applied. (Currently the auto-style format bricks other checks from running.)

See the very last commits and #2028 for usage example. (Sorry for the terrible commit messages.)

### Test Case 1
1. Pull this branch
2. Make a style error (e.g. add an extra space to mess up indentation)
3. Wait a few minutes, see if the auto-fix PR pops up and tags me (due to GH limitations I can only set it to tag the PR author)

**Expected result:** A PR by GitHub Actions bot should be created, tagging both this PR and its author (me)

### Test Case 2
1. After Test Case 1, merge the auto-fix PR in for me ("squash and merge")

**Expected result:** The style fixes will be applied, and build-and-test should re-run and appear at the bottom of this PR. (This can take a while.)

### CODE REVIEW

File: soccer/src/soccer/planning/planner_node.hpp [hunk 1]
```diff
@@ -75,8 +75,10 @@ class SharedStateInfo {
             });
     }
 
+    // TODO: do these need mutex locks?
     [[nodiscard]] PlayState play_state() const {
-        auto lock = std::lock_guard(mutex_);
+        auto lOCK = std::lock_guard(mutex_);
+
         return last_play_state_;
```
Comment (kasohrab) [inline soccer/src/soccer/planning/planner_node.hpp]: Yes it should have to be locked: https://preshing.com/20130618/atomic-vs-non-atomic-operations/.

**Review by kasohrab — CHANGES_REQUESTED** [review]
You can delete the todo. Basically since it takes more than one cpu cycle it could be race condition. cpp is not a high enough level language to abstract the need for locking away (like python has for the most part)

**Comment by kasohrab** [comment]
Also the rest looks great! We should port it to firmware.

**Comment by kasohrab** [comment]
Oh I realize that todo was also for testing lol

**Comment by kfu02** [comment]
> Oh I realize that todo was also for testing lol

yes lol but it was a genuine question I had too, so thanks!

**Comment by kfu02** [comment]
> Also the rest looks great! We should port it to firmware.

not quite, it still doesn't run when the bot does [auto-style](https://github.com/RoboJackets/robocup-software/pull/2023/commits/f091df4cce047bfd225b005001597aae1297781c), let me work on that

**Review by N8BWert — APPROVED** [review]
worked for me

**Comment by dwu359** [comment]
Works

**Review by dwu359 — APPROVED** [review]

**Review by kasohrab — APPROVED** [review]
Lgtm

**Pull request merged**